### DISCUSSION TITLE
Releases

### DISCUSSION THREAD

**Discussion by fade2gray** [mark-wiemer/ahkpp#46] [Q&A] [answered]
AutoHotkey-Plus has a Marketplace release, it also has several forks of which AutoHotkey-Plus-Plus is one but is the only one that also has a Marketplace release.

Is it possible to release a fork of AutoHotkey-Plus-Plus? I have a use-case where I would like to have two separate instances of AHK++ (AHK++ and a clone) installed in VSC.

**Comment by fade2gray** [comment]
I think I've found a workaround for my use-case by using VSC installed and VSC portable, so I can have two separate workspaces using different settings for extensions.

https://medium.com/@allancarneirosantos/how-to-use-multi-installation-of-vscode-a69de1955161

**Comment by mark-wiemer** [comment, marked as answer]
If your workaround doesn't work, you can always clone the repo, change the package name, package the repo, and install your VSIX locally without publishing (or waiting for someone else to publish). Again, you would have to manually update, but it should be easy enough to write a script to automate this process. That way you won't be switching between VS Code windows (one regular and one portable), you'll just be enabling/disabling extensions within one window. Not sure which fits your workflow better :)

**Reply by fade2gray** [reply]
> you can always clone the repo, change the package name, package the repo, and install your VSIX locally

Got there eventually. 🍺

![snip-1](https://user-images.githubusercontent.com/1332009/108104040-b0405e80-7082-11eb-84df-f53e1b9bf247.png)
![snip-2](https://user-images.githubusercontent.com/1332009/108104056-b3d3e580-7082-11eb-900e-29022179f8cd.png)

**Comment by mark-wiemer** [comment]
I'm curious, why do you need this use-case? Is this just for fixing the Run Selection categorization or is it for something else as well?

**Reply by fade2gray** [reply]
@mark-wiemer 
> Is this just for fixing the Run Selection categorization

No, I was just highlighting that I discovered the command needed recategorizing after my initial attempt at renaming the package.

My use-case is to find a method that assists in making script conversions a little easier, e.g. A32 to U64 or U64(v1) to U64(2) etc., and the need for the original script to launch from its editor tab using one ahk executable and a copy of the script to be launched from another tab using a different executable - remember one of my first PRs was to extend the settings menu to include all ahk executables making them selectable by mouseclick from a dropdown menu?

Then I discovered the significance of using separate Workspaces which I further enhanced by having them open in separate monitors but of course, I still had to toggle between executables to see the effect of any changes I'd made to one script over the other because of the shared `settings.json`  - that's why I took the route of creating a local clone of AHK++.

...but while I found it a great experience creating and repackaging that clone, I quickly discovered this which I hadn't paid any attention to before...

![snip-3](https://user-images.githubusercontent.com/1332009/108202961-c9481e80-7119-11eb-93e6-586bfe87e310.png)

...so, I can now change AHK++ settings to my heart's content for each saved Workspace without the need for a clone. 😎
